test/includes/microcloud.sh

```diff
@@ -271,8 +271,8 @@ set_debug_binaries() {
     lxc exec "${name}" -- rm -f /var/snap/microcloud/common/microcloudd.debug
     lxc exec "${name}" -- rm -f /var/snap/microcloud/common/microcloud.debug
 
-    lxc file push --quiet "${MICROCLOUDD_DEBUG_PATH}" "${name}"/var/snap/microcloud/common/microcloudd.debug
-    lxc file push --quiet "${MICROCLOUD_DEBUG_PATH}" "${name}"/var/snap/microcloud/common/microcloud.debug
+    lxc file push --quiet "${MICROCLOUDD_DEBUG_PATH}" "${name}"/var/snap/microcloud/common/microcloudd.debug --mode 0755
```

Comment:
Out of interest what is the reason for changing the permission for the debug binaries?

Reply:
`lxc file push` sets the file mode to `0644` by default - we want to make sure the binaries are executable and maintain the file mode set by the Go compiler. We also check for the presence of these executables in `commands/daemon.start` and `commands/microcloud` in the MicroCloud snap. When they are present and executable, the `GOCOVERDIR` is created. Here's the relevant PR: https://github.com/canonical/microcloud-pkg-snap/pull/15.

Reply:
You can set mode with file push

https://documentation.ubuntu.com/lxd/en/latest/reference/manpages/lxc/file/push/

Reply:
Are you sure LXD changes to permissions to 0644 by default? From what I have observed it picks up the same permissions as they are present on the host.

Reply:
LXD doesn't change the permissions to 0644 by default unless it can't grab the host file permissions. When I initially tested this, the `GOCOVERDIR` wasn't being created and I also didn't see the echo'd message "==> WARNING: Using a custom debug MicroCloud binary!". Upon further investigation, it looks like the upload artifact GitHub action doesn't maintain permissions: https://github.com/actions/upload-artifact?tab=readme-ov-file#permission-loss. We can either tar the binaries together prior to uploading the artifact or continue using `lxc file push` with the mode flag.

Reply:
I see ok, thanks for checking!

Reply:
No problem!